app/browser/tabs.js

```diff
@@ -619,6 +623,87 @@ const api = {
       api.createTab(state, action)
     }
     return state
+  },
+
+  goBack: (state, action) => {
+    action = makeImmutable(action)
+    const tab = api.getWebContents(action.get('tabId'))
+    if (tab && !tab.isDestroyed()) {
+      tab.goBack()
+    }
+    return state
+  },
+
+  goForward: (state, action) => {
+    action = makeImmutable(action)
+    const tab = api.getWebContents(action.get('tabId'))
+    if (tab && !tab.isDestroyed()) {
+      tab.goForward()
+    }
+    return state
+  },
+
+  goToIndex: (state, action) => {
+    action = makeImmutable(action)
+    const tab = api.getWebContents(action.get('tabId'))
+    if (tab && !tab.isDestroyed()) {
+      tab.goToIndex(action.get('index'))
+    }
+    return state
+  },
+
+  getWindowId: () => {
+    if (BrowserWindow.getFocusedWindow()) {
```

Comment:
I think this should probably be a `getActiveWindowId` method on windows.js and return `windowState.WINDOW_ID_NONE` instead of -1 just for consistency